python-package/lightgbm/callback.py

```diff
@@ -50,19 +50,27 @@ def _format_eval_result(value: list, show_stdv: bool = True) -> str:
 
 
 def print_evaluation(period: int = 1, show_stdv: bool = True) -> Callable:
```

Comment:
maybe we could rename this to `log_evaluation` as well? Seems like it is only used in `engine` right now so it could be a good time before it is used explicitly as a callback.

Reply:
@jmoralez 

Good idea, I like it! Unfortunately, `print_evaluation()` callback is already in public API, so we cannot simply rename it.
https://lightgbm.readthedocs.io/en/latest/Python-API.html#callbacks

I'll prepare a separate PR for that if you don't mind.